cmd/kubeadm/app/util/apiclient.go

```diff
@@ -52,3 +54,29 @@ func WaitForAPI(client *clientset.Clientset) {
 		return true, nil
 	})
 }
+
+// WaitForPodsWithLabel will lookup pods with the given label and wait until they are all
+// reporting status as running.
+func WaitForPodsWithLabel(client *clientset.Clientset, name, labelKeyValPair string) {
+	wait.PollInfinite(kubeadmconstants.APICallRetryInterval, func() (bool, error) {
+		listOpts := metav1.ListOptions{LabelSelector: labelKeyValPair}
+		apiPods, err := client.CoreV1().Pods(metav1.NamespaceSystem).List(listOpts)
+		if err != nil {
+			fmt.Printf("[apiclient] Error getting %s pods [%v]\n", name, err)
+			return false, nil
+		}
+		fmt.Printf("[apiclient] Found %d %s pods\n", len(apiPods.Items), name)
```

Comment:
is it worth outputting when 0 pods are found? if not, can we move this beneath the `len(apiPods.Items) == 0` check